src/protocol/server.c

```diff
@@ -92,17 +90,50 @@ protocolServerError(ProtocolServer *this, int code, const String *message, const
     ASSERT(message != NULL);
     ASSERT(stack != NULL);
 
-    KeyValue *error = kvNew();
-    kvPut(error, VARSTR(PROTOCOL_ERROR_STR), VARINT(code));
-    kvPut(error, VARSTR(PROTOCOL_OUTPUT_STR), VARSTR(message));
-    kvPut(error, VARSTR(PROTOCOL_ERROR_STACK_STR), VARSTR(stack));
+    // Write the error and flush to be sure it gets sent immediately
+    PackWrite *error = pckWriteNew(this->write);
+    pckWriteU32P(error, protocolMessageTypeError);
+    pckWriteI32P(error, code);
+    pckWriteStrP(error, message);
+    pckWriteStrP(error, stack);
+    pckWriteEndP(error);
 
-    ioWriteStrLine(protocolServerIoWrite(this), jsonFromKv(error));
-    ioWriteFlush(protocolServerIoWrite(this));
+    ioWriteFlush(this->write);
 
     FUNCTION_LOG_RETURN_VOID();
 }
 
+/**********************************************************************************************************************************/
+ProtocolServerCommandGetResult
+protocolServerCommandGet(ProtocolServer *const this)
```

Comment:
Is this the getter for `protocolCommandPut()` - because the logic isn't symmetrical so I'm trying to find the getter for `protocolCommandPut()`...